azure/converters/subnets.go

```diff
@@ -17,21 +17,17 @@ limitations under the License.
 package converters
 
 import (
-	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
+	asonetworkv1 "github.com/Azure/azure-service-operator/v2/api/network/v1api20201101"
 	"k8s.io/utils/ptr"
 )
 
-// GetSubnetAddresses returns the address prefixes contained in an SDK v2 subnet.
-func GetSubnetAddresses(subnet *armnetwork.Subnet) []string {
+// GetSubnetAddresses returns the address prefixes contained in an ASO subnet.
+func GetSubnetAddresses(subnet asonetworkv1.VirtualNetworksSubnet) []string {
 	var addresses []string
-	if subnet.Properties != nil && subnet.Properties.AddressPrefix != nil {
-		addresses = []string{ptr.Deref(subnet.Properties.AddressPrefix, "")}
-	} else if subnet.Properties != nil && subnet.Properties.AddressPrefixes != nil {
-		for _, address := range subnet.Properties.AddressPrefixes {
-			if address != nil {
-				addresses = append(addresses, *address)
-			}
-		}
+	if subnet.Status.AddressPrefix != nil {
+		addresses = []string{ptr.Deref(subnet.Status.AddressPrefix, "")}
+	} else if subnet.Status.AddressPrefixes != nil {
+		addresses = subnet.Status.AddressPrefixes
```

Comment:
> 1. If yes, shall we be considering either one of it? Is there a reason to choose `subnet.Status.AddressPrefix` over `subnet.Status.AddressPrefixes` ?

I'm not entirely sure, but I was trying to keep this behavior the same in this PR so I think changing it should be done separately.

> 2. Can we append both if `subnet.Status.AddressPrefix` and `subnet.Status.AddressPrefixes` are different ?

I don't think this would ever happen. My guess is that both exist only because it started out as the singular field and then the API evolved to support multiple but kept the old field for compatibility.

Reply:
> My guess is that both exist only because it started out as the singular field and then the API evolved

That's correct

AddressPrefixes was added to support dual stack clusters (one ipv4 cidr and one ipv6). Originally only single stack ipv4 was supported. In the Azure SDK the same two fields exist and only one can be set.